src/application.hpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2019-2021 Ashar Khan <ashar786khan@gmail.com>
+ *
+ * This file is part of CP Editor.
+ *
+ * CP Editor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * I will not be responsible if CP Editor behaves in unexpected way and
+ * causes your ratings to go down and or lose any important contest.
+ *
+ * Believe Software is "Software" and it isn't immune to bugs.
+ *
+ */
+
+#ifndef APPLICATION_HPP
+#define APPLICATION_HPP
+
+#include <QFileOpenEvent>
+#include <singleapplication.h>
+
+class Application : public SingleApplication
```

Comment:
Whats the intention of bringing new class. Can't you use existing SingleApplication class?

Reply:
How to Override event method without inheriting?

Reply:
> How to Override event method without inheriting?

Are you asking me?

Reply:
I mean there was no other way around except inherit to override

Reply:
See this: https://doc.qt.io/qt-5/qfileopenevent.html

Reply:
I didnt know that there is no singleapplication.hpp file. I thought we could reuse SingleApplication class. 😀😀😀